### ISSUE TITLE
[Bug]: Action-button with href cannot be activated by screen reader

### ISSUE THREAD

**Issue by jnurthen** [adobe/spectrum-web-components#4576]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-action-button, sp-button

### Expected behavior

buttons should be able to be activated by the screen reader

### Actual behavior

Buttons with href cannot be activated when in screen reader browse mode

### What browsers are you seeing the problem in?

Chrome, Safari, Microsoft Edge

### How can we reproduce this issue?

1. Go to https://studio.webcomponents.dev/edit/m5wlfZcQ5coVjJcjkUjE/src/index.ts?p=stories
2. Turn on screen reader
3. Navigate to the first action button
4. Try to activate the button
5. Nothing happens
6. Same with the 3rd button

### Sample code that illustrates the problem

https://studio.webcomponents.dev/edit/m5wlfZcQ5coVjJcjkUjE/src/index.ts?p=stories

### Logs taken while reproducing problem

https://github.com/adobe/spectrum-web-components/assets/1724479/3cc4d8a1-c250-4e4e-9a91-a5d862c24d9e

https://github.com/adobe/spectrum-web-components/assets/1724479/ffe1270d-4f92-401f-aa7c-98a43a4c2e46
